Add exllamav2 better (#27111)
huggingface/transformers: docs/source/en/main_classes/quantization.md

### Exllama kernels for faster inference

For 4-bit model, you can use the exllama kernels in order to a faster inference speed. It is activated by default. You can change that behavior by passing `use_exllama` in [`GPTQConfig`]. This will overwrite the quantization config stored in the config. Note that you will only be able to overwrite the attributes related to the kernels. Furthermore, you need to have the entire model on gpus if you want to use exllama kernels. 

```py
import torch

> gptq_config = GPTQConfig(bits=4)

Reviewer: I'm guessing this is missing? 

```suggestion
gptq_config = GPTQConfig(bits=4, use_exllama=True)
```

Author: The default value is `True`, so we don't need to pass it.

> model = AutoModelForCausalLM.from_pretrained("{your_username}/opt-125m-gptq", device_map="auto", quantization_config = gptq_config)

Reviewer: nit: pep8

```suggestion
model = AutoModelForCausalLM.from_pretrained("{your_username}/opt-125m-gptq", device_map="auto", quantization_config=gptq_config)
```

### Exllama kernels for faster inference

For 4-bit model, you can use the exllama kernels in order to a faster inference speed. It is activated by default. You can change that behavior by passing `use_exllama` in [`GPTQConfig`]. This will overwrite the quantization config stored in the config. Note that you will only be able to overwrite the attributes related to the kernels. Furthermore, you need to have the entire model on gpus if you want to use exllama kernels. Also, you can perform CPU inference using Auto-GPTQ for Auto-GPTQ version > 0.4.2 by passing `device_map` = "cpu". For CPU inference, you have to pass `use_exllama = False` in the `GPTQConfig.`

```py
import torch
gptq_config = GPTQConfig(bits=4)
model = AutoModelForCausalLM.from_pretrained("{your_username}/opt-125m-gptq", device_map="auto", quantization_config=gptq_config)
```

> With the release of the exllamav2 kernels, you can get faster inference speed compared to the exllama kernels. You just need to 
> pass `exllama_version=2` in [`GPTQConfig`] and disable exllama kernels:

Reviewer: ```suggestion
With the release of the exllamav2 kernels, you can get faster inference speed compared to the exllama kernels. You just need to pass `exllama_config={"version": 2}` in [`GPTQConfig`]:
```

```py
import torch

> gptq_config = GPTQConfig(bits=4, exllama_version={"version":2})

Reviewer: ```suggestion
gptq_config = GPTQConfig(bits=4, exllama_config={"version":2})
```
